### PR TITLE
JBPM-6748: Stunner - Improve domain model declaration by making some annotations not mandatory.

### PR DESCRIPTION
**Pull request by romartin**
Hey @manstis @jomarko @hasys 

As this week we're doing kinda workshop, I'd been showcasing the Stunner domain beans and did some cleans that were missing from time ago, this way you will see the code for beans and properties has been reduced.

This commit provides no fixes neither new stuff, it's just a code improvement, so nothing special to test here rather than it works as usually I guess... :) Added some missing tests too.

See [JBPM-6748](https://issues.jboss.org/browse/JBPM-6748)

It **depends on** 
- https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/672
- https://github.com/kiegroup/kie-wb-distributions/pull/670
- https://github.com/kiegroup/jbpm-wb/pull/971/

Thanks!

### CODE REVIEW

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by jomarko** [comment]
jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/BaseThrowingIntermediateEvent.java [hunk 1]
```diff
@@ -78,7 +78,9 @@ protected void initLabels(){
         labels.add("EventOnChoreographyActivityBoundary");
         labels.add("IntermediateEventsMorph");
         labels.add("cmnop");
-    };
+    }
+
+    ;
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/BaseThrowingIntermediateEvent.java]: Minor comment.. formatting has shown the floating ```;```!

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/main/java/org/kie/workbench/common/stunner/core/i18n/AbstractTranslationService.java [hunk 2]
```diff
@@ -45,7 +44,7 @@ public String getDefinitionSetDescription(String defSetId) {
 
     @Override
     public String getPropertySetName(String proepSetId) {
-        return getKeyValue(proepSetId + I18N_SEPARATOR + PROPERTY_SET_NAME_SUFFIX);
+        return getKeyValue(proepSetId + I18N_SEPARATOR + CAPTION_SUFFIX);
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/main/java/org/kie/workbench/common/stunner/core/i18n/AbstractTranslationService.java]: While I am reviewing.. please note typo in parameter name (```propSetId``` no ```e```)

**Comment by manstis** [comment]
@romartin Looks OK to me.

I assume support for "optional", "read-only" , ```@Title``` and ```@Description``` have been removed!? 

Were these no longer required?

**Comment by romartin** [comment]
@manstis yep, two things:
- I realized almost all properties were setting both `optional` and `readOnly` as false, so just changed this annotations in order to be not mandatory, if present, you can set the values to `true`, otherwise will default to `false`.
- About title and description the same, you can set the annotations, but by default now Stunner only looks in the 18n bundle for the properties (by using some naming convention), so there is no need to add literals in both properties and beans at this point.
Let you know that tomorrow will try to improve this a bit more with  @pefernan , so probably we'll add a new commit on top.
Thanks!!

**Comment by romartin** [comment]
Hey @hasys !! Well it does not remove any property from the domain.. not sure if following you sorry. This PR is about doing some cleans on the domain declaration (classes) - But not removing properties from the domain, just removing metadata from the properties/nodes. For example I mean no need for setting the title, desc, etc for a given property bean (by annotating some field) if we're already creating the literals in some i18n bundle, it's redundant, isn't it?.So things like that... Does this makes sense/answers your Q? :) 
Lemme know whatever!
Thx!

**Comment by hasys** [comment]
Hi @romartin,

yes, sorry, you are right, my bad. This PR is ready to merge, I guess.

**Comment by romartin** [comment]
hehe np  @hasys !!! 
Well hold on guys, I'm looking at this with @pefernan , probably we'll add a new commit on top of this to make the model even more clear... will let you know.
So please **do not merge yet**, let's keep this PR open yet for a while.
Thanks!!

**Comment by jomarko** [comment]
No problem for me, please, just let us know when it is again ready for review.

**Comment by romartin** [comment]
Hey @hasys @jomarko @pefernan 
Finally were able to set the `@FieldLabel` annotation as not mandatory, so updated model beans again, as the label is already being given by the i18n bundle.
Sorry for big commit finally - more than 200 files updated, but if you realize if because did lots of cleans in all model classes, so not expecting  it a problematic commit although its size... :) 
This is ready for review again. Thanks!

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Added a few commits for the missing i18n translations

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Pull request merged**